homeassistant/config_entries.py

```diff
@@ -3064,6 +3064,19 @@ def _async_abort_entries_match(
             match_dict,
         )
 
+    @property
+    def _config_entry_id(self) -> str:
+        """Return config entry id."""
+        # This is the same as handler, but that's an implementation detail
+        return self.handler
+
+    @callback
+    def _get_config_entry(self) -> ConfigEntry:
```

Comment:
I think I would prefer this to be a property actually. I don't think we can get here without the handler being a proper valid entity id?

Reply:
I followed the same pattern as ConfigFlow (reauth/reconfigure)
It does a lookup which could theoretically fail (even if it shouldn't), in case the entry has been removed in another browser or something like that.

I would prefer to leave it as a method.

Reply:
In the `ConfigFlow` it was a bit different as there was nothing and also could be in different context e.g. source.
In the `OptionsFlow` (actually `OptionsFlowWithConfigEntry`) there is already something e.g. a property.

By using a property it's a drop in replacement in the various config flows in integrations to change `OptionsFlowWithConfigEntry` to just `OptionsFlow` instead of also needing to modify the rest of the code to retrieve the config entry. Protecting that the config entry still exist could go into `__init__`.

Anyhow, maybe to get a second opinion here.

Reply:
> Protecting that the config entry still exist could go into `__init__`.

I don't understand your comment about this.
- If one wants to reference the config_entry from `__init__`, then it needs to be passed in the constructor (can't use the default method/property).
- If one doesn't care about the config_entry in `__init__`, then it doesn't need to be passed in and we should not encourage users to pass it in.

Reply:
I'd prefer a property inside `OptionsFlow` to return `ConfigEntry`. An options flow can't exist without a config entry. It's not the same case as in the config flow where flows will have different sources.

Reply:
The config entry can be passed to the options flow by the integration if the flow needs an init method and config entry. If it doesn't need an init method it will have the config entry available in the default property.

Reply:
Having it as a property causes havoc either the existing flows... unless we find a name that's unlikely to be already used

Reply:
OK, I think I have it working by using a compatibility/temporary property setter.
That should prevent the hard breaking change.

Reply:
Please confirm if you approve this path, before I work on the extra tests and on the blog post...